pkg/cri/opts/spec_linux.go

```diff
@@ -306,6 +330,24 @@ func WithDevices(osi osinterface.OS, config *runtime.ContainerConfig) oci.SpecOp
 				return err
 			}
 		}
+
+		if enableDeviceOwnershipFromSecurityContext {
+			UID := getDeviceUserGroupID(config.GetLinux().GetSecurityContext().GetRunAsUser())
+			GID := getDeviceUserGroupID(config.GetLinux().GetSecurityContext().GetRunAsGroup())
+			// Loop all new devices added by oci.WithDevices() to update their
+			// dev.UID/dev.GID.
+			//
+			// non-zero UID/GID from SecurityContext is used to override host's
+			// device UID/GID for the container.
+			for idx := range s.Linux.Devices[oldDevices:] {
+				if UID != 0 {
+					*s.Linux.Devices[idx].UID = UID
```

Comment:
The idx is starting from 0. If the oldDevice is not zero, it doesn't work, right?

Reply:
@fuweid right, of course!

Reply:
```
-                        for idx := range s.Linux.Devices[oldDevices:] {
+                       for idx := oldDevices; idx < len(s.Linux.Devices); idx++ {
```

Reply:
good catch @fuweid